apps/namada-interface/src/slices/accounts.ts

```diff
@@ -194,48 +201,90 @@ const balancesAtom = (() => {
       set(base, {});
 
       // Split accounts into transparent and shielded
-      const [transparentAccounts, shieldedAccounts] = accounts.reduce((acc, curr) => {
-        if (curr.isShielded) {
-          acc[1].push(curr)
-        } else {
-          acc[0].push(curr)
-        }
-        return acc;
-      }, [[], []] as [AccountDetails[], AccountDetails[]]);
+      const [transparentAccounts, shieldedAccounts] = accounts.reduce(
+        (acc, curr) => {
+          if (curr.isShielded) {
+            acc[1].push(curr);
+          } else {
+            acc[0].push(curr);
+          }
+          return acc;
+        },
+        [[], []] as [AccountDetails[], AccountDetails[]]
+      );
 
       const token = nativeToken || tokenAddress;
 
       // We query the balances for the transparent accounts first as it's faster
-      const transparentBalances = await Promise.all(queryBalance(namada, transparentAccounts, token));
-      transparentBalances.forEach(([address, balance ]) => {
+      const transparentBalances = await Promise.all(
+        queryBalance(namada, transparentAccounts, token)
+      );
+      transparentBalances.forEach(([address, balance]) => {
         set(base, { ...get(base), [address]: balance });
       });
 
       await namada.sync();
 
-      const shieldedBalances = await Promise.all(queryBalance(namada, shieldedAccounts, token));
-      shieldedBalances.forEach(([address, balance ]) => {
+      const shieldedBalances = await Promise.all(
+        queryBalance(namada, shieldedAccounts, token)
+      );
+      shieldedBalances.forEach(([address, balance]) => {
         set(base, { ...get(base), [address]: balance });
       });
     }
   );
 })();
 
 const queryBalance = (
-  int: InstanceType<Integration>,
+  namada: Namada,
   accounts: AccountDetails[],
-  token: string,
-): Promise<readonly [string, Balance]>[] => {
-  return accounts.map(async (account) => {
-    const result = await int.queryBalances(account.address, [token]);
-    return [
-      account.address,
-      result.reduce(
-        (acc, curr) => ({ ...acc, [curr.token]: new BigNumber(curr.amount) }),
-        {} as Balance,
-      ),
-    ] as const;
+  token: string
+): Promise<[string, TokenBalances]>[] => {
+  return accounts.map(async (account): Promise<[string, TokenBalances]> => {
+    const balances = await namada.queryBalances(account.address, [token]);
+    return [account.address, balances];
   });
 };
 
-export { accountsAtom, balancesAtom };
+const keplrAccountAtom = (() => {
+  const base = atom(new Promise<AccountDetails | undefined>(() => {}));
+
+  return atom(
+    (get) => get(base),
+    (_get, set) =>
+      set(
+        base,
+        (async () => {
+          const keplr = getIntegration("cosmos");
+          const accounts = await keplr.accounts();
+
+          // keplr only supports one account
+          return accounts?.[0];
```

Comment:
Not for this PR specifically, but it would be better if we invoke Keplr with something like(inside the Keplr integration), as this _will_ actually give us access to all accounts at once, assuming the user has approved connections for those chains and they are available in the Keplr extension:
```typescript
(async () => {
  // Define a constant somewhere for all supported chains
  const chains = ["cosmoshub-1", "theta-testnet-1", "osmosis-1"];
  await keplr.enable(chains); // Needs to be invoked when we actually connect Keplr
  const accounts = await Promise.all(chains.map(async (chainId) => {
    return keplr.getKey(chainId);
  }));
    console.log(accounts);
})();
```
We can do that later though, then here we can just return accounts as it is returned from integration!

Reply:
Ah, I see! I think I'm getting confused about what an account is on Keplr. Is it correct to say:
- Within Keplr you can have multiple wallets, of which one can be selected at any time
- Each wallet holds multiple addresses (one per chain ID)
- When we call `keplr.getKey(chainID)`, we are requesting the address information for a specific chain ID on the currently selected wallet
- `signer.getAccounts()` just returns the address information for one chain ID for some reason?

In which case I need to fix the jotai atom to be an array of accounts. I got confused by this in the Keplr docs :weary::

>You can get the address/public keys by `getAccounts` method.
It can return the array of address/public key.
But, currently, Keplr extension manages only one address/public key pair.

Reply:
yeah it's weird in Keplr - you can derive custom paths, and I think if you do you will end up with multiple addresses for one chain ID, but I need to confirm that it still does that. I think there will be a use-case for enabling and querying accounts from all chains, but unfortunately that will involve a lot of approval pop-ups for every chain enabled, at least the first time a user connects. You can leave it for now, but either way we should probably use `getKeys` so we have something to display for the account name/alias.